bcryptjs/index.d.ts

```diff
@@ -63,13 +70,16 @@ export declare function compareSync(s: string, hash: string): boolean;
  * @param  hash             Data to be compared to
  * @param  callback         Callback receiving the error, if any, otherwise the result
  * @param  progressCallback Callback successively called with the percentage of rounds completed (0.0 - 1.0), maximally once per MAX_EXECUTION_TIME = 100 ms.
+ * @return Promise, if callback has been omitted
+ * @throws If callback is present but not a function
  */
-export declare function compare(s: string, hash: string, callback: (err: Error, success: boolean) => void, progressCallback?: (percent: number) => void): void;
+export declare function compare(s: string, hash: string, callback?: (err: Error, success: boolean) => void, progressCallback?: (percent: number) => void): Promise<boolean>;
 
 /**
  * Gets the number of rounds used to encrypt the specified hash.
  * @param  hash Hash to extract the used number of rounds from
  * @return Number of rounds used
+ * @throws If hash is not a string
```

Comment:
This is what type annotations are for. Most functions can throw an exception if they get a parameter of the wrong type.